Marlin/src/module/stepper.cpp

```diff
@@ -2818,7 +2820,8 @@ hal_timer_t Stepper::block_phase_isr() {
 
       // Calculate the initial timer interval
       interval = calc_multistep_timer_interval(current_block->initial_rate << oversampling_factor);
-      acceleration_time += interval;
+      // Initialize ac/deceleration time as if half the time passed.
+      acceleration_time = deceleration_time = interval / 2;
```

Comment:
In retrospect I am questioning this line. It will mean that after the first step of the segment the speed will have increased by half what it should have.

I think the effect you were aiming for is something like:
```
      interval = calc_multistep_timer_interval(current_block->initial_rate << oversampling_factor);
      acceleration_time = deceleration_time = interval / 2;
      acceleration_time += interval;
```

That would be more consistent with the deceleration ramp half-initialisation where `deceleration_time` is approximately half a step ahead so the calculated speed for a segments is an estimate of the average speed of the segment rather than the speed at the start of the segment.

An even more consistent way of doing it would be to have initialised `acceleration_time` to `ticks_nominal / 2` in the previous deceleration ramp and use that to estimate an average speed for the first segment. However the `minimum_planner_speed` logic basically does this for us.

Reply:
> It will mean that after the first step of the segment the speed will have increased by half what it should have.

Half acceleration/deceleration from first step (which is always at `initial_rate`) to the second step is absolutely intentional. The effect of this was highlighted with green checkmarks in my logic analyzer capture.
The other intended effect is that for full acceleration/deceleration segments we (often) won't hit the `final_rate` exactly on the last step. Rather, it will be off by half-ac/deceleration. Which is a great speed difference to have as the `final_rate` of current segment is equal to `initial_rate` of next segment.

Why? 2 things:
1) To reduce (in most cases) the acceleration at the transition between segments. These transition points are crucial because at them we have extra jerk if 2+ axes are involved. The extra jerk is due to cornering with jerk control or junction deviation.
2) It is a simple solution to ensure that we keep speed changes to close 1x the acceleration limit in the various transitions between all the possible cases of segments: full accel / full decel / accel into (/cruise into) decel / accel into cruise / cruise into decel. The close to 1x the acceleration limit adds from 0.5x from first-to-second step acceleration plus or minus up to 0.5x from last-to-first step ac/deceleration (because we won't hit the `final_rate` exactly). Rounding into discrete number of ac/deceleration steps affects this as well.

> the effect you were aiming for is something [..]

That would result in 1.5x acceleration for first-to-second step speed chage.